### PR TITLE
Addressed metadata documentation cleanup from #589

### PR DESCRIPTION
**Pull request by fehguy**
See https://github.com/OAI/OpenAPI-Specification/issues/589#issuecomment-250816162

### CODE REVIEW

**Comment by fehguy** [comment]
Fixes #199, #377

**Comment by darrelmiller** [comment]
In the OpenAPI object the `hosts` key is still present but in the example it is shown as `servers`.  Do we have a preference?

Also the description of the Host Object seems to have got stuck in the middle of the description of the Info Object.

**Comment by DavidBiesack** [comment]
I prefer `servers` over `hosts`. The example shows 
```
servers:
-   url: https://development.gigantic-server.com/v1
```
and I think "server" (as in "API server" / "service") is a better term here than host, as `host` should refer to just the  host name `development.gigantic-server.com`. I don't think we should create a new definition of `host` that is not compatible with the [definition of host in a URL](https://tools.ietf.org/html/rfc3986#section-3.2.2)

**Comment by fehguy** [comment]
@darrelmiller i will fix that.

**Comment by darrelmiller** [comment]
Need to add a new doc that discusses migration guidelines and expectations from tooling vendors in regards to being compatible between minor versions.

File: versions/3.0.md [hunk 1]
```diff
@@ -196,9 +205,73 @@ An object representing a Host.
...
+<a name="hostTemplates"></a>templates | [Templates Object](#hostTemplatesObject) | An object holding templates for substitution in the URL template
+
+This object can be extended with [Specification Extensions](#specificationExtensions). 
+
+##### Host Object Example
+
+The following shows how multiple hosts can be described in the host object array
+
+```yaml
+servers:
```
Comment (earth2marsh) [inline versions/3.0.md]: alternatives for `servers`: `instances` … ?

**Comment by fehguy** [comment]
the terms `hosts` and `servers` need to be updated accordingly in the operations / paths objects

File: versions/3.0.md [hunk 2]
```diff
@@ -196,9 +205,73 @@ An object representing a Host.
...
+- url: {scheme}://{host}:{port}/{basePath}
+  description: The production API server
+  templates:
+    scheme:
+      enum:
+        - https
+        - http
+      default: https
```
Comment (earth2marsh) [inline versions/3.0.md]: `default` currently applies to optional parameters and represents the server-assumed default, like the number of results in a page or an offset. By applying it to a required element, we confuse an already tricky aspect of a spec. An alternative could be `example`.

File: versions/3.0.md [hunk 3]
```diff
@@ -196,9 +205,73 @@ An object representing a Host.
...
+  description: Production server
+```
+
+The following shows how templates an be used for a host configuration
+
+```yaml
+servers:
+- url: {scheme}://{host}:{port}/{basePath}
+  description: The production API server
+  templates:
```
Comment (earth2marsh) [inline versions/3.0.md]: As discussed, introducing `templates` when other path templates use `parameters` seems less elegant, but I don't have a better alternative to supply at this time.

**Comment by fehguy** [comment]
@webron to add documentation about what happens when the `servers` attribute is empty or not present

**Pull request merged**